test_responses.py

```diff
@@ -390,3 +396,56 @@ def run():
 
     run()
     assert_reset()
+
+
+def test_response_timeout():
+    body = b'test'
+    status = 200
+    url = 'http://example.com/'
+    timeout = 0.1  # timeout in seconds
+
+    @responses.activate
+    def run():
+        responses.add(responses.GET, url, body=body, timeout=timeout)
+        begin = time.time()
+        resp = requests.get(url)
+        end = time.time()
+        assert end - begin >= timeout  # some additional delay may took place
+        assert resp.text == 'test'
+        assert resp.status_code == status
+
+    run()
+    assert_reset()
+
+
+def test_request_timeout():
+    status = 200
+    url = 'http://example.com/'
+    timeout = 0.1  # timeout in seconds
+
+    @responses.activate
+    def run():
+        responses.add(responses.GET, url, timeout=timeout)
+        begin = time.time()
+        resp = requests.get(url, timeout=timeout)
+        end = time.time()
+        assert end - begin >= timeout
+        assert resp.status_code == status
+
+    run()
+    assert_reset()
+
+
+def test_request_timeout_expired():
+    url = 'http://example.com/'
+    response_timeout = 0.2
+    request_timeout = 0.1
+
+    @responses.activate
+    def run():
+        responses.add(responses.GET, url, timeout=response_timeout)
```

Comment:
You could get the same result by using `body=ConnectTimeout` here. No need for a new `timeout` argument. https://github.com/getsentry/responses/blob/master/responses.py#L243